neard/src/cli.rs

```diff
@@ -93,10 +96,17 @@ impl NeardCmd {
             NeardSubCommand::RecompressStorage(cmd) => {
                 cmd.run(&home_dir);
             }
-        }
+        };
+        Ok(())
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+pub(crate) enum RunError {
+    #[error("Invalid EnvFilter directives provided")]
```

Comment:
conventionally the error messages should not be capitalized. They also should be aimed at the user, and so should refer to user-facing concepts, not developer facing-ones

```suggestion
    #[error("invalid logging directives provided")]
```